### ISSUE TITLE
Usage of loss_dummy

### ISSUE THREAD

**Issue by NikhilMank** [nish03/FFS#3]
Hi @nish03, I have a question regarding the usage of loss_dummy. I understand why loss_dummy1 since there was no computation performed that involves weight energy but the logistic_regression network is already involved in the calculation of lr_reg_loss then why calculate loss_dummy again for the same network?

In loss_dummy1, is self.weight_energy.bias already 0? According to my understanding, if it is not zero, loss_dummy1 should be some very small value but not 0. Same with the loss_dummy, is the logistic_regression.bias 0?

**Comment by nish03** [comment]
Hi @NikhilMank thanks for the question. This again boils down to Detectron2 functionality, where it is mandatory to explicitly define losses for every learnable weights. Check Line 544 and Line 559 where `torch.nn.Linear` is defined for `logistic_regression` and `weight_energy`, due to which it is essential to separately express loss for it. You can check the values of `loss_dummy` and `loss_dummy1` at each iteration in the training log file.

**Comment by NikhilMank** [comment]
Since `logistic regression` network is already involved in the calculation of `lr_reg_loss`, I didn't understand the need for creating `loss_dummy`

I have seen log_dir and both `loss_dummy` and `loss_dummy1` was always 0. That is why I was asking if the bias is made 0. I didn't find anything while initialization that made the `bias=0` or `False` but since both `loss_dummy` and `loss_dummy1` are zero throughout the training, it is only possible when the `bias=0`. I don't know if this actually affects the training or not but I was just curious about this.

**Comment by nish03** [comment]
I do recall that when I commented out `loss_dummy` and/or `loss_dummy1`, detectron2 throw me an error. Do let me know if that happens in your case as well.

**Comment by KostadinovShalon** [comment]
Hi all, I'm not completely sure tbh, but I remember looking at the same dummy losses when I was working with the original VOS repo, and I remember everything was fine while training on single GPU, however it seemed that it was needed for multiple-gpu training, otherwise the training would fail while starting to train the uncertainty-related weights.
